### PR TITLE
[flutter_tools] remove mocks from android emulator tests

### PR DESCRIPTION
**Pull request by jonahwilliams**
Work towards #71511

### CODE REVIEW

**Comment by jonahwilliams** [comment]
oops, these are real failures - investigating

**Comment by jonahwilliams** [comment]
Fixed

File: packages/flutter_tools/test/general.shard/android/android_emulator_test.dart [hunk 1]
```diff
@@ -161,20 +151,14 @@ void main() {
             exitCode: 1,
             stderr: errorText,
             stdout: 'dummy text',
-            duration: Duration(seconds: 1),
+            duration: Duration(milliseconds: 1),
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/android/android_emulator_test.dart:154]: I'm guessing with async code, `Duration.zero == Duration(milliseconds: 1)`?

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/android/android_emulator_test.dart:154]: it means "at least zero", but probably more

**Comment by jonahwilliams** [comment]
Still need to debug this one I guess - passes reliably locally but I might just be covering up some races

File: packages/flutter_tools/test/general.shard/android/android_emulator_test.dart [hunk 2]
```diff
@@ -188,21 +172,20 @@ void main() {
             exitCode: 1,
             stderr: '',
             stdout: 'dummy text',
-            duration: Duration(seconds: 4),
+            duration: Duration(milliseconds: 50),
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/android/android_emulator_test.dart:175]: Is this deterministic? where does 50 come from?

**Comment by jonahwilliams** [comment]
I think this can work without the durations. One of the tests needs to just not exit before we run the assertion, we can use a completer to keep it up forever. The other doesn't need to take any time at all

**Comment by jonahwilliams** [comment]
(of course lets see what CI says)

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**